### PR TITLE
Add prepareMethod=none execution path and tests

### PR DESCRIPTION
**Pull request by divang**
**Summary**
This PR introduces a new prepareMethod=none connection property option that uses literal parameter substitution + SQL Batch execution for all prepared statements.

When prepareMethod=none is set, the driver always uses literal parameter substitution + SQL Batch for ALL prepared statements, bypassing server-side prepared statement handles (sp_prepexec/sp_prepare). This provides server-side visibility for performance tuning and query analysis.

**Changes**
Added NONE("none") to PrepareMethod enum in SQLServerDriver.java
Added isPrepareMethodNone execution path in SQLServerPreparedStatement.java

**Tests**
**Basic Functionality Tests:**
testBasicOperationsWithPrepareMethodNone - Tests basic CRUD operations with direct SQL execution
testBatchExecutionWithPrepareMethodNone - Validates batch insert operations work correctly
testMultipleDataTypesWithPrepareMethodNone - Tests literal parameter substitution for INT, BIGINT, DECIMAL, FLOAT, VARCHAR, NVARCHAR, BIT, DATE, DATETIME2
testNullValuesWithPrepareMethodNone - Ensures NULL values are correctly substituted in SQL
testSelectQueryWithPrepareMethodNone - Tests parameterized SELECT queries with LIKE and comparison operators

**SQL Batch Execution Tests:**
testBatchMultipleParameterizedStatementsWithNone - Multiple INSERT statements with parameters in single batch
testBatchNonParameterizedBetweenParameterizedWithNone - Non-parameterized statement (DELETE) between parameterized INSERTs
testBatchMultipleParametersPerStatementWithNone - Verifies correct parameter-to-position mapping across multiple statements
testBatchOnlyNonParameterizedStatementsWithNone - Batch execution with no parameters
testBatchWithStringLiteralsAndCommentsWithNone - SQL parsing edge cases - semicolons in string literals, comments
testBatchParameterMappingCorrectnessWithNone - Validates exact parameter values are inserted in correct tables/columns
testBatchCombinedWithErrorsWithNone - Error handling - duplicate key violation throws SQLException
testLargeBatchWithNone - Scalability test with 100 batch statements

### CODE REVIEW

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java [hunk 1]
```diff
@@ -841,8 +842,8 @@ public final class SQLServerDriver implements java.sql.Driver {
                     SQLServerDriverStringProperty.SERVER_CERTIFICATE.getDefaultValue(), false, null),
             new SQLServerDriverPropertyInfo(SQLServerDriverStringProperty.PREPARE_METHOD.toString(),
                     SQLServerDriverStringProperty.PREPARE_METHOD.getDefaultValue(), false,
-            new String[] { PrepareMethod.PREPEXEC.toString(), PrepareMethod.PREPARE.toString(),
-                    PrepareMethod.SCOPE_TEMP_TABLES_TO_CONNECTION.toString() }),
+                new String[] { PrepareMethod.NONE.toString(), PrepareMethod.PREPEXEC.toString(),
```
Comment (machavan) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java]: extra starting space

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 2]
```diff
@@ -1130,7 +1133,8 @@ private boolean reuseCachedHandle(boolean hasNewTypeDefinitions, boolean discard
             return false;
 
         // No caching for exec method as it always executes directly without preparation
-        if (connection.getPrepareMethod().equals(PrepareMethod.SCOPE_TEMP_TABLES_TO_CONNECTION.toString())) {
+        if (connection.getPrepareMethod().equals(PrepareMethod.SCOPE_TEMP_TABLES_TO_CONNECTION.toString())
```
Comment (machavan) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: We have a bug with this check for SCOPE_TEMP_TABLES_TO_CONNECTION. Just the presence of this preparemethod does not mean we will be using direct sql execution and return false from here. 

I think we may have to preserve isDirectSqlRequest  value in a member variable in PrepaedStatement object and use if across functions.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 3]
```diff
@@ -1230,16 +1234,16 @@ private boolean doPrepExec(TDSWriter tdsWriter, Parameter[] params, boolean hasN
...
 
-        // If using scopeTempTablesToConnection method, check if temp tables are present
-        // in the SQLisScopeTempTablesToConnectionns
-        if (isScopeTempTablesToConnection && containsTemporaryTableOperations(userSQL)) {
-            // Build direct SQL using enhanced replaceParameterMarkers with direct values
-            String directSQL = replaceParameterMarkersWithValues(userSQL, userSQLParamPositions, params,
-                    false);
+        // For prepareMethod=none, always send direct SQL. For prepareMethod=scopeTempTablesToConnection,
+        // send direct SQL only when temporary table operations are detected in the user SQL.
+        if (isPrepareMethodNone || (isScopeTempTablesToConnection && containsTemporaryTableOperations(userSQL))) {
```
Comment (machavan) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: here also we can just leverage the isDirectSqlRequest  flag if we preserve that in a member variable

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PrepStmtDirectSQLExecutionTests.java [hunk 4]
```diff
@@ -1356,4 +1356,777 @@ public void testBatchCombinedWithErrors() throws Exception {
...
+     * Test Case: Batch with errors using prepareMethod=none
+     * Verifies update counts reflect failures correctly with literal parameter
+     * substitution
+     */
+    @Test
+    public void testBatchCombinedWithErrorsWithNone() throws Exception {
+        String tempTable = "#temp_none_errors_" + ThreadLocalRandom.current().nextInt(1000, 9999);
+
+        try (SQLServerConnection conn = (SQLServerConnection) PrepUtil.getConnection(connectionString)) {
+            conn.setPrepareMethod("none");
```
Comment (machavan) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PrepStmtDirectSQLExecutionTests.java:2124]: have some test that doesn't use this method but directly put the prepareMethod on connection string.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 5]
```diff
@@ -133,6 +133,16 @@ private void setPreparedStatementHandle(int handle) {
...
 
+    /**
+     * Flag indicating whether this execution will use direct SQL execution
+     * (PKT_QUERY)
+     * instead of prepared statements (PKT_RPC). This is true when:
+     * - prepareMethod=none (always direct SQL), or
+     * - prepareMethod=scopeTempTablesToConnection AND temp table operations are
+     * detected
+     */
+    private boolean isDirectSqlExecution = false;
```
Comment (machavan) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: We can populate the flag in SQLServerPreparedStatement constructor by analyzing the userSQL and use it in both the single and batch execution paths.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 6]
```diff
@@ -300,6 +308,23 @@ private boolean resetPrepStmtHandle(boolean discardCurrentCacheItem) {
         userSQLParamPositions = parsedSQL.parameterPositions;
         initParams(userSQLParamPositions.length);
         useBulkCopyForBatchInsert = conn.getUseBulkCopyForBatchInsert();
+
+        // Calculate prepare method flags once in constructor since prepareMethod and
```
Comment (machavan) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java:312]: format comments

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 7]
```diff
@@ -300,6 +308,23 @@ private boolean resetPrepStmtHandle(boolean discardCurrentCacheItem) {
...
+        // userSQL
+        // don't change after construction
+        String prepareMethod = connection.getPrepareMethod();
+        boolean isScopeTempTables = prepareMethod.equals(PrepareMethod.SCOPE_TEMP_TABLES_TO_CONNECTION.toString());
```
Comment (machavan) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]:
```
if (prepareMethod is NONE){
  isDirectSqlExecution = true
  usePrepExec = false
} else if (prepareMethod is scopeTempTables) {
  if (containsTemporaryTableOperations(userSQL)) {
      isDirectSqlExecution = true
     usePrepExec = false
  } else {
      isDirectSqlExecution = false
     usePrepExec = true
  }
} else if (prepareMethod is preexec) {
  usePrepExec = true
}```

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java [hunk 8]
```diff
@@ -290,6 +308,31 @@ private boolean resetPrepStmtHandle(boolean discardCurrentCacheItem) {
...
+                usePrepExec = false;
+            } else {
+                isDirectSqlExecution = false;
+                usePrepExec = true;
+            }
+        } else if (prepareMethod.equals(PrepareMethod.PREPEXEC.toString())) {
+            isDirectSqlExecution = false;
+            usePrepExec = true;
+        } else {
+            // Default: prepareMethod=prepare (sp_prepare)
```
Comment (machavan) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java]: Thie // default comment is a bit misleading since it seems to say that the default prepare method is prepare. 

Lets just remove this comment

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PrepStmtDirectSQLExecutionTests.java [hunk 9]
```diff
@@ -1356,4 +1356,962 @@ public void testBatchCombinedWithErrors() throws Exception {
...
+            }
+
+            // Verify data types preserved
+            try (PreparedStatement ps = conn.prepareStatement("SELECT * FROM " + tableName);
+                    ResultSet rs = ps.executeQuery()) {
+                assertTrue(rs.next());
+                assertEquals(Integer.MAX_VALUE, rs.getInt("col_int"));
+                assertEquals(Long.MAX_VALUE, rs.getLong("col_bigint"));
+                assertEquals(new BigDecimal("123456789.123456"), rs.getBigDecimal("col_decimal"));
+                assertEquals("ASCII string", rs.getString("col_varchar"));
```
Comment (muskan124947) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PrepStmtDirectSQLExecutionTests.java:1629]: can we add assert statement for missing columns too

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PrepStmtDirectSQLExecutionTests.java [hunk 10]
```diff
@@ -1356,4 +1356,962 @@ public void testBatchCombinedWithErrors() throws Exception {
             }
         }
     }
+
+    // ==================== prepareMethod=none Tests ====================
```
Comment (muskan124947) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PrepStmtDirectSQLExecutionTests.java:1359]: Can we move all the resource cleanup in finally block

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PrepStmtDirectSQLExecutionTests.java [hunk 11]
```diff
@@ -1356,4 +1356,969 @@ public void testBatchCombinedWithErrors() throws Exception {
             }
         }
     }
+
```
Comment (machavan) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PrepStmtDirectSQLExecutionTests.java:1358]: Can we rename this file name now to say something like PrepStmtDirectSQLExecutionTests ?

File: src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PrepStmtDirectSQLExecutionTests.java [hunk 12]
```diff
@@ -38,13 +38,12 @@
...
- * implementation
- * Focus: Temp table persistence, parameter
- * substitution, security, edge cases, and performance
+ * Comprehensive test suite for PreparedStatement direct SQL execution methods.
+ * Tests scopeTempTablesToConnection and prepareMethod=none implementations.
+ * Focus: Temp table persistence, parameter substitution, security, edge cases, and performance.
  */
 @RunWith(JUnitPlatform.class)
-public class PrepareMethodScopeTempTablesToConnectionTest extends AbstractTest {
+public class PrepStmtDirectSQLExecutionTests extends AbstractTest {
```
Comment (Ananya2) [inline src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PrepStmtDirectSQLExecutionTests.java:46]: All the prepareMethod=none tests here are temp-table (#temp) based. Can we add one simple non-#temp (regular table) DML/SELECT test to validate literal substitution works for general queries too, not just temp-table scenarios?
